Source/WarpXIO.cpp

```diff
@@ -570,87 +567,87 @@ WarpX::UpdateInSitu () const
 #ifdef _OPENMP
 #pragma omp parallel reduction(+:n_per_proc)
 #endif
-                for (MFIter mfi(mf[lev]); mfi.isValid(); ++mfi) {
+                for (MFIter mfi(mf[lev]); mfi.isValid(); ++mfi)
                     n_per_proc += npart_in_grid[mfi.index()];
-                }
-                mf[lev].setVal(static_cast<Real>(n_per_proc), dcomp,1);
+
+                mf[lev].setVal(static_cast<Real>(n_per_proc), dcomp, ngrow);
+
                 if (lev == 0)
-                {
                     varnames.push_back("part_per_proc");
-                }
+
                 dcomp += 1;
             }
         }
 
         if (plot_proc_number)
         {
+            Real procid = static_cast<Real>(ParallelDescriptor::MyProc());
```

Comment:
Could you explain why you convert `int ParallelDescriptor::MyProc()` to `Real`?

Reply:
Hi sure, I copied this code from plot file code. I think it's because it's easier to store all data reals, instead of some arrays as real and some integer.